packages/SwingSet/src/kernel/slogger.js

```diff
@@ -130,36 +137,33 @@ export function makeSlogger(slogCallbacks, writeObj) {
         console.error(
           `WARNING: slogger state confused: vat ${vatID} in ${state}, not ${exp}: ${msg}`,
         );
-        write({ type: 'slogger-confused', vatID, state, exp, msg });
+        safeWrite({ type: 'slogger-confused', vatID, state, exp, msg });
       }
     }
 
     function vatConsole(sourcedConsole) {
-      const vc = {};
-      for (const level of ['debug', 'log', 'info', 'warn', 'error']) {
-        vc[level] = (sourceTag, ...args) => {
-          if (replay) {
-            // Don't duplicate stale console output.
-            return;
-          }
-          sourcedConsole[level](sourceTag, ...args);
-          const when = { state, crankNum, vatID, deliveryNum };
-          const source = sourceTag === 'ls' ? 'liveslots' : sourceTag;
-          write({ type: 'console', source, ...when, level, args });
-        };
-      }
-      return harden(vc);
+      return makeLimitedConsole(level => (source, ...args) => {
+        // Don't duplicate stale output.
+        if (replay) return;
+
+        // Write to the console, then to the slog.
+        sourcedConsole[level](source, ...args);
+        // TODO: Just use "liveslots" rather than "ls"?
+        if (source === 'ls') source = 'liveslots';
```

Comment:
It's weird that we map source "ls" to "liveslots" in the slog but not in console output. For that matter, it's weird that we use an abbreviation just to reverse it anyway.

Reply:
we have some documented `DEBUG` config mentioning `ls` as the anylogger prefix, which I believe is used for being concise. However from what I gather we want the slog to have `liveslots` as source.

I suspect the abbreviation is the main case so that the emitter can also use the abbreviated form.